File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int right = height.length - 1;
        int left = 0;
        int max = 0;
        while(left < right){
            int sum  = Math.min(height[left], height[right]) * (right - left);
            if(sum > max) max = sum;
            if(height[left] > height[right]) right--;
            else left++;
        }
        return max;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int max1 = 0;
        while(left < right){
            int area = min(height[left], height[right]) * (right - left);
            if(max1 < area) max1 = area;
            if(height[left] < height[right]) left++;
            else right--;
        }
        return max1;
    }
};